### PR TITLE
add provider checking based on model name and provider

### PR DESCRIPTION
**Pull request by PCSwingle**

### CODE REVIEW

File: mentat/llm_api_handler.py [hunk 1]
```diff
@@ -142,19 +155,48 @@ async def initialize_client(self):
...
+                from mentat.session_input import collect_user_input
+
+                match provider.name:
+                    case "open_ai" | "openai":
+                        env_variable = "OPENAI_API_KEY"
+                    case "anthropic":
+                        env_variable = "ANTHROPIC_API_KEY"
+                    case "azure":
+                        if os.getenv("AZURE_OPENAI_ENDPOINT") is None:
+                            raise MentatError(
```
Comment (jakethekoenig) [inline mentat/llm_api_handler.py:183]: I think we should prompt the user to enter the endpoint too.

File: mentat/llm_api_handler.py [hunk 2]
```diff
@@ -142,19 +155,48 @@ async def initialize_client(self):
...
+                            f"No api key detected for provider {provider.name}. Create a .env file in ~/.mentat/.env or in your workspace root with your api key"
+                        )
+
+                ctx.stream.send(
+                    f"No {provider.name} api key detected. To avoid entering your api key on startup, create a .env file in"
+                    " ~/.mentat/.env or in your workspace root.",
+                    style="warning",
+                )
+                ctx.stream.send("Enter your api key:", style="info")
+                key = (await collect_user_input(log_input=False)).data
```
Comment (jakethekoenig) [inline mentat/llm_api_handler.py:198]: If I enter q here the program hangs

**Review by jakethekoenig — COMMENTED** [review]
When I set my model to "test" and my provider to "openai" or "open_ai" I get the same exception
image

**Comment by jakethekoenig** [comment]
This might not be totally connected to this PR but it would be nice if mentat didn't disable the user input box if they get a 404 from the api. For instance if I set my model to test I get the following (reasonable) error but then I can't enter anything into the box again
image

**Comment by PCSwingle** [comment]
> This might not be totally connected to this PR but it would be nice if mentat didn't disable the user input box if they get a 404 from the api.

Agreed but it might be better to do in a separate PR

**Review by jakethekoenig — APPROVED** [review]
Looks good to me. I guess another thing to do in another PR is upgrade to 0.2 spice

**Pull request merged**